src/color.js

```diff
@@ -366,34 +365,40 @@ function translateColor (ycc, luma) {
 
   let intersection = null;
 
-  for (let i = 0; i < cubeFaces.length; i++) {
-    let cubeFace = cubeFaces[i];
-    intersection = findIntersection(line, cubeFace);
+  let cubeFaces = (lighter ? YCC_CUBE_FACES_WHITE : YCC_CUBE_FACES_BLACK);
+  for (let cubeFace of cubeFaces) {
+    let candidateIntersection = findIntersection(line, cubeFace);
+    if (candidateIntersection.z < 0 || candidateIntersection.z > 1)
+      continue;
```

Comment:
At first I thought this was reworking line 373 from the before column, but I'm unable to follow the logic here.